src/ProductConstructionService/ProductConstructionService.Client/ProductConstructionServiceApiOptions.cs

```diff
@@ -28,8 +42,13 @@ public ProductConstructionServiceApiOptions(string baseUri, string accessToken,
         /// <param name="managedIdentityId">Managed Identity to use for the auth</param>
         public ProductConstructionServiceApiOptions(string accessToken, string managedIdentityId)
             : this(
-                  new Uri(ProductConstructionServiceApi.StagingPcsBaseUri),
-                  ProductConstructionServiceApi.CreateApiCredential(ProductConstructionServiceApi.StagingPcsBaseUri, accessToken, managedIdentityId))
+                  new Uri(StagingPcsBaseUri),
+                  AppCredentialResolver.CreateTokenCredential(
+                      EntraAppIds[StagingPcsBaseUri.TrimEnd('/')],
+                      disableInteractiveAuth: true, // services should not support interactive authentication
```

Comment:
This comment is technically wrong. This is a client for the PCS service. Can be used from `darc` too.
We only use it in Maestro so far so we don't need the interactive credential for now. But might in the future.

Reply:
Oh, I see. Thanks for the correction, will fix